plugins/language_vala.lua

```diff
@@ -0,0 +1,121 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Vala",
+  files = { "%.vala$" },
+  comment = "//",
+  block_comment = { "/*", "*/" },
+  patterns = {
+    { pattern = "//.*",                                                           type = "comment"  }, -- Single-string comment
+    { pattern = { "/%*", "%*/" },                                                 type = "comment"  }, -- Multi-line comment
+    { pattern = { '"', '"', '\\' },                                               type = "string"   }, -- ?
+    { pattern = { "'", "'", '\\' },                                               type = "string"   }, -- ?
+    { pattern = "-?0x%x+",                                                        type = "number"   }, -- ?
+    { pattern = "-?%d+[%d%.eE]*f?",                                               type = "number"   }, -- ?
+    { pattern = "-?%.?%d+f?",                                                     type = "number"   }, -- ?
+    { regex   = "\\w+(?=\\s+\\w++\\s+\\=\\s+)",                                   type = "function" }, -- Class name in class instance
+    { regex   = "\\<.+\\>",                                                       type = "keyword2" }, -- Generic Type
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",                                           type = "operator" }, -- ?
+    { pattern = "[%a_][%w_]*%f[(]",                                               type = "function" }, -- ?
+    { pattern = "[%a_][%w_]*",                                                    type = "symbol"   }, -- ?
+    { regex   = "\\s?\\:\\s?(?:\\w+\\.?)+(?=\\s?[{])",                            type = "keyword2" }, -- Inheritance
+    { regex   = "\\s?\\:\\s?(?:\\w+(?:\\<.+\\>)?\\s?\\,?\\s?)+(?=\\s?[{])",       type = "keyword2" }, -- Inheritance
+    { regex   = "class()\\s\\w+()\\<.+\\>(?=\\s?\\:)",                            type = { "keyword", "normal", "keyword2" } }, -- Generic Class Type
+    { regex   = "interface()\\s\\w+()\\<.+\\>(?=\\s?\\:)",                        type = { "keyword", "normal", "keyword2" } }, -- Generic Interface Type
+    { regex   = "\\=\\>(?=[{])",                                                  type = "keyword"  }, -- Lambda
+    { regex   = "[A-Z][A-Z_]+",                                                   type = "keyword2" }, -- Constants
+    { regex   = "\\[.+\\](?=\\w*)",                                               type = "keyword2" }, -- Attribute
+    { regex   = "\\#[if|elif|else|endif](?=\\s?\\w*)",                            type = "keyword"  }, -- Preprocessor directive
```

Comment:
I don't think this regex is correct.
`[if|elif|else|endif]` will not match the words `if`, `elif`, `else` and `endif`.

Reply:
I changed the pattern, thanks.